generators/ruby/sdk/src/ClientsGenerator.ts

```diff
@@ -374,6 +421,41 @@ export class ClientsGenerator {
         const typeExporterLocation = "types_export";
         clientFiles.push(new GeneratedRubyFile({ rootNode: typeExporter, fullPath: typeExporterLocation }));
 
+        let oauthTokenProvider: OauthTokenProvider | undefined;
+        if (this.shouldGenerateOauth && this.oauthScheme != null) {
+            const accessTokenClass = new AccessToken(this.clientName, this.oauthScheme.configuration);
+
+            let maybeRefereshTokenDetails;
+            if (this.oauthScheme.configuration.refreshEndpoint) {
+                maybeRefereshTokenDetails = getOauthRefreshTokenFunctionMetadata({
+                    refreshEndpoint: this.oauthScheme.configuration.refreshEndpoint,
+                    artifactRegistry: this.artifactRegistry
+                });
+            }
+            const fullRubyDetails = {
```

Comment:
nit: maybe best to just name this `oauthConfiguration` so its consistent with the property name used below?